### ISSUE TITLE
Cannot load filters from disk in branch setup_cleanup

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#26]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/26) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

If you try to compute_filters() on a data object where they already have been computed and stored in HDF5 file, for some reason the code tries to construct a Filter() object anyway and uses an unassigned variable name:

```
#!python


UnboundLocalError                         Traceback (most recent call last)
<ipython-input-1-abd318541669> in <module>()
----> 1 import ppp_paper as ppp; data=ppp.load(); ds=data.channel[209]; g=ds.good()

/Users/fowlerj/Microcal/Tupac_analysis/Riken_calibration_metals/ppp_paper.pyc in load(channels, massname, reset_bad_channels)
     77     data.avg_pulses_auto_masks()
     78     data.compute_noise_spectra(1000)
---> 79     data.compute_filters(f_3db=5000)
     80     data.filter_data()
     81     data.drift_correct()

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mass/core/channel_group.pyc in compute_filters(self, fmax, f_3db, forceNew)
   1139                 print("chan %d skipping compute_filter because already done, and loading filter" % ds.channum)
   1140                 h5grp = ds.hdf5_group['filters']
-> 1141                 ds.filter = Filter(avg_signal, self.nPresamples-ds.pretrigger_ignore_samples,
   1142                                    spectrum, ds.noise_autocorr, sample_time=self.timebase,
   1143                                    fmax=fmax, f_3db=f_3db, shorten=2)

UnboundLocalError: local variable 'avg_signal' referenced before assignment


```

Can you restore this to its former behavior? Can't merge this to master if it won't load filters from HDF5 file! @oneilg  might be able to help.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

p.s. Note that I committed a small change this morning about the .gitignore file. You should pull that before working on this.

**Comment by joefowler** [comment]
**Original comment by Young Il Joe (Bitbucket: [jojo978](https://bitbucket.org/jojo978), GitHub: [jojo978](https://github.com/jojo978)).**

----------------------------------------

I fixed this problem (avg_signal => ds.average_pulse, spectrum => ds.noise_spectrum). I tested this and it worked but I don't understand why it worked before.

**Comment by joefowler** [comment]
**Original comment by Young Il Joe (Bitbucket: [jojo978](https://bitbucket.org/jojo978), GitHub: [jojo978](https://github.com/jojo978)).**

----------------------------------------

But it doesn't check that average pulses are already calculated and stored in the HDF5 file. Maybe I should put more dependency check there.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Okay, now I have a new one.
```
#!python

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-abd318541669> in <module>()
----> 1 import ppp_paper as ppp; data=ppp.load(); ds=data.channel[209]; g=ds.good()

/Users/fowlerj/Microcal/Tupac_analysis/Riken_calibration_metals/ppp_paper.pyc in load(channels, massname, reset_bad_channels)
     77     data.avg_pulses_auto_masks()
     78     data.compute_noise_spectra(1000)
---> 79     data.compute_filters(f_3db=5000)
     80     data.filter_data()
     81     data.drift_correct()

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mass/core/channel_group.pyc in compute_filters(self, fmax, f_3db, forceNew)
   1140                 h5grp = ds.hdf5_group['filters']
   1141                 ds.filter = Filter(ds.average_pulse[...], self.nPresamples-ds.pretrigger_ignore_samples,
-> 1142                                    ds.noise_spectrum[...], ds.noise_autocorr[...], sample_time=self.timebase,
   1143                                    fmax=fmax, f_3db=f_3db, shorten=2)
   1144                 ds.filter.peak_signal = h5grp.attrs['peak']

AttributeError: 'MicrocalDataSet' object has no attribute 'noise_spectrum'

```

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Bug fix in instantiating Filter from HDF5 data. Fixes #26

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Bug fix in instantiating Filter from HDF5 data. Fixes #26
